cloudant.js

```diff
@@ -29,67 +29,35 @@ var reconfigure = require('./lib/reconfigure.js');
 function Cloudant(options, callback) {
   debug('Initialize', options);
 
-  // Save the username and password for potential conversion to cookie auth.
-  var login = reconfigure.getOptions(options);
-
-  // Convert the credentials into a URL that will work for cloudant. The
-  // credentials object will become squashed into a string, which is fine
-  // except for the .cookie option.
-  var cookie = options.cookie;
-
-  var pkg = require('./package.json');
-  var requestDefaults = {
-    gzip: true,
-    headers: {
-      // set library UA header
-      'User-Agent': `nodejs-cloudant/${pkg.version} (Node.js ${process.version})`
-    },
-    jar: false
-  };
-  var theurl = null;
-  if (typeof options === 'object') {
-    if (options.requestDefaults) {
-      requestDefaults = options.requestDefaults;
-    }
-    theurl = reconfigure(options);
-  } else {
-    theurl = reconfigure({ url: options });
+  if (typeof options !== 'object') {
+    options = { url: options };
   }
+
+  var theurl = reconfigure(options);
   if (theurl === null) {
+    var err = new Error('Invalid URL');
```

Comment:
Is there a test case for invalid URL?

Reply:
This is asserted in the legacy reconfigure tests.